### DISCUSSION TITLE
'Diff' plot style issue

### DISCUSSION THREAD

**Discussion by maxpaton** [kdbinsidebrains/plugin#106] [General]
Hi there,

I used your 'diff' plot style (instead of line, etc) for the first time today and noticed it didn't look right in places. Am I right in thinking it should essentially be a fill between two series? But it seems to be filling correctly for some records but not others.

Here is a reproducible example:

/ plotting this
update bid:mid-5,ask:mid+5 from ([] time:2024.10.22+03:00+til 10;mid:50 50 70 70 70 70 70 70 90 90)
![image](https://github.com/user-attachments/assets/b637e51c-cbdd-4dd4-ad58-47970d13be5b)
The result when configured with a line (mid) and diff between bid and ask (white), we can see from 3:00-3:01 and 03:08-03:09, the mid  isn't half way between the bid and ask as I believe it should be - or am I misunderstanding the use of the 'diff' plot style?

Thanks a lot!
Max

**Comment by smklimenko** [comment]
when you create two series, each seria has own Y-axis range. If you look at the left Y-axis called Value (unfortunately, you use same name for both, but it should be the axis for Diff series), you can see it's range from 45 to 97. On the right side you can Y-axis for another Value (that should be for Mid values) that is from 48 to 92. You see correct mid value here only by one reason: both series have middle value 70.

Series are designed to put differ scale values into one graph, for example, volumes and prices. To have same Y-ranges for some values you should use the same series.
